src/Data/Nat/Binary/Minus.agda

```diff
@@ -0,0 +1,364 @@
+  fromℕ ((k ℕ.+ m) ℕ.∸ n)      ≡⟨ cong fromℕ (ℕₚ.+-∸-assoc k n≤m) ⟩
+  fromℕ (k ℕ.+ (m ℕ.∸ n))      ≡⟨ fromℕ-homo-+ k (m ℕ.∸ n) ⟩
+  fromℕ k + fromℕ (m ℕ.∸ n)    ≡⟨ cong (_+ (fromℕ (m ℕ.∸ n))) (fromℕ-toℕ x) ⟩
+  x + fromℕ (m ℕ.∸ n)          ≡⟨ cong (x +_) (fromℕ-homo-∸ m n) ⟩
+  x + (fromℕ m ∸ fromℕ n)      ≡⟨ cong (x +_) $ cong₂ _∸_ (fromℕ-toℕ y) (fromℕ-toℕ o) ⟩
+  x + (y ∸ o)                  ∎
+  where
+  open ≤-Reasoning
+  k = toℕ x;  m = toℕ y;  n = toℕ o;  n≤m = toℕ-mono-≤ o≤y
+
+x+y∸x+z :  ∀ x y z →  (x + y) ∸ (x + z) ≡ y ∸ z
+x+y∸x+z x y z =  begin-equality
+  (x + y) ∸ (x + z)    ≡⟨ sym (∸-+ (x + y) x z) ⟩
+  ((x + y) ∸ x) ∸ z    ≡⟨ cong (_∸ z) ([x+y]∸x≡y x y) ⟩
+  y ∸ z                ∎
+  where open ≤-Reasoning
+
+suc-x∸suc-y :  ∀ x y → suc x ∸ suc y ≡ x ∸ y
+suc-x∸suc-y x y =  begin-equality
+  suc x ∸ suc y         ≡⟨ cong₂ _∸_ (suc≗1+ x) (suc≗1+ y) ⟩
+  (1ᵇ + x) ∸ (1ᵇ + y)   ≡⟨ x+y∸x+z 1ᵇ x y ⟩
+  x ∸ y                 ∎
+  where open ≤-Reasoning
+
+*-distribˡ-∸ :  FP-Bin._DistributesOverˡ_ _*_ _∸_
+*-distribˡ-∸ x y z =  begin-equality
+  x * (y ∸ z)                         ≡⟨ sym (fromℕ-toℕ (x * (y ∸ z))) ⟩
+  fromℕ (toℕ (x * (y ∸ z)))           ≡⟨ cong fromℕ (toℕ-homo-* x (y ∸ z)) ⟩
+  fromℕ (k ℕ.* (toℕ (y ∸ z)))         ≡⟨ cong (fromℕ ∘ (k ℕ.*_)) (toℕ-homo-∸ y z) ⟩
+  fromℕ (k ℕ.* (m ℕ.∸ n))             ≡⟨ cong fromℕ (ℕₚ.*-distribˡ-∸ k m n) ⟩
+  fromℕ ((k ℕ.* m) ℕ.∸ (k ℕ.* n))     ≡⟨ fromℕ-homo-∸ (k ℕ.* m) (k ℕ.* n) ⟩
+  fromℕ (k ℕ.* m) ∸ fromℕ (k ℕ.* n)   ≡⟨ cong₂ _∸_ (fromℕ-homo-* k m)
+                                                   (fromℕ-homo-* k n) ⟩
+    (fromℕ k * fromℕ m) ∸ (fromℕ k * fromℕ n)
+                                     ≡⟨ cong₂ _∸_ (cong₂ _*_ (fromℕ-toℕ x) (fromℕ-toℕ y))
+                                                  (cong₂ _*_ (fromℕ-toℕ x) (fromℕ-toℕ z)) ⟩
+    x * y ∸ x * z                    ∎
+  where
+  open ≤-Reasoning;  k = toℕ x;  m = toℕ y;  n = toℕ z
+
+*-distribʳ-∸ :  _*_ FP-Bin.DistributesOverʳ _∸_
+*-distribʳ-∸ x y z =  begin-equality
+  (y ∸ z) * x     ≡⟨ *-comm _ x ⟩
+  x * (y ∸ z)     ≡⟨ *-distribˡ-∸ x y z ⟩
+  x * y ∸ x * z   ≡⟨ cong₂ _∸_ (*-comm x y) (*-comm x z) ⟩
+  y * x ∸ z * x   ∎
+  where open ≤-Reasoning
+
+∸-mono-≤ :  _∸_ Preserves₂ _≤_ ⟶ _≥_ ⟶ _≤_
+∸-mono-≤ {x} {x'} {y} {y'} x≤x' y'≤y =  begin
+  x ∸ y                   ≡⟨ sym (fromℕ-toℕ (x ∸ y)) ⟩
+  fromℕ (toℕ (x ∸ y))     ≡⟨ cong fromℕ (toℕ-homo-∸ x y) ⟩
+  fromℕ (xN ℕ.∸ yN)       ≤⟨ fromℕ-mono-≤ (ℕₚ.∸-mono xN≤x'N y'N≤yN) ⟩
+  fromℕ (x'N ℕ.∸ y'N)     ≡⟨ fromℕ-homo-∸ x'N y'N ⟩
+  fromℕ x'N ∸ fromℕ y'N   ≡⟨ cong₂ _∸_ (fromℕ-toℕ x') (fromℕ-toℕ y') ⟩
+  x' ∸ y'                 ∎
+  where
+  open ≤-Reasoning
+  xN = toℕ x;   yN = toℕ y;   x'N = toℕ x';   y'N = toℕ y'
+
+  xN≤x'N = toℕ-mono-≤ x≤x';   y'N≤yN = toℕ-mono-≤ y'≤y
+
+∸-monoˡ-≤ :  (x : ℕᵇ) → (_∸ x) Preserves _≤_ ⟶ _≤_
+∸-monoˡ-≤ x y≤z =  ∸-mono-≤ y≤z (≤-refl {x})
+
+x∸y≤x :  ∀ x y → x ∸ y ≤ x
+x∸y≤x x y =  begin
+  x ∸ y    ≤⟨ ∸-mono-≤ (≤-refl {x}) (0≤x y) ⟩
```

Comment:
You could split this proof out to the right version of `∸-monoˡ-≤`

Reply:
Done.